src/server.rs

```diff
@@ -171,37 +178,49 @@ struct Opts {
     /// location to store backup files
     #[clap(short, long)]
     dir: Option<String>,
-    /// alternative port location
+    /// alternative uds location
     #[clap(short, long)]
-    port: Option<std::net::SocketAddr>,
+    uds: Option<String>,
 }
 
 #[tokio::main]
 #[allow(unused)]
 pub async fn main() -> Result<(), Box<dyn std::error::Error>> {
-    let default_grpc_port = "[::1]:10000".parse().unwrap();
+    let default_uds_path = "/var/tmp/nv-dhcp".to_string();
     let default_config_dir = "".to_string();
 
     env_logger::builder().format_timestamp(None).init();
     let opts = Opts::parse();
 
     // where we store the cache file
-    let conf_dir = opts.dir.unwrap_or_else(|| default_config_dir);
+    let conf_dir = opts.dir.unwrap_or(default_config_dir);
     // location of the grpc port
-    let addr = opts.port.unwrap_or_else(|| default_grpc_port);
+    let uds_path = opts.uds.unwrap_or(default_uds_path);
+
+    tokio::fs::create_dir_all(Path::new(&uds_path).parent().unwrap()).await?;
+    let uds = UnixListener::bind(&uds_path)?;
+    let uds_stream = UnixListenerStream::new(uds);
 
     let cache = match LeaseCache::new(None) {
         Ok(c) => Arc::new(Mutex::new(c)),
         Err(e) => {
-            log::error!("IO error with the cache fs");
+            log::error!("Could not setup the cache: {}", e.to_string());
             return Ok(());
         }
     };
-
     let netavark_proxy_service = NetavarkProxyService(cache);
     Server::builder()
         .add_service(NetavarkProxyServer::new(netavark_proxy_service))
-        .serve(addr)
+        .serve_with_incoming(uds_stream)
         .await?;
+
+    //Clean up UDS on exit
+    return match fs::remove_file(uds_path) {
+        Ok(_) => Ok(()),
+        Err(e) => {
+            warn!("Could not clear the uds file: {}", e);
```

Comment:
s/clear/remove s/uds//

Reply:
Can you elaborate on what you mean here?

Reply:
replace clear with remove and remove 'uds'